src/algorithms/diamondsquare.jl

```diff
@@ -0,0 +1,258 @@
+"""
+    DiamondSquare
+
+This type generates a neutral landscape using the diamond-sqaures
+algorithm, which produces fractals with variable spatialautocorrelation.
+
+https://en.wikipedia.org/wiki/Diamond-square_algorithm
+
+The algorithm is named diamond-square because it is an iterative procedure of
+"diamond" and "square" steps.
+
+
+
+The degree of spatial autocorrelation is controlled by a parameter `H`,
+which varies from 0.0 (low autocorrelation) to 1.0 (high autocorrelation) --- note this is non-inclusive and H = 0 and H = 1 will not behavive as expected.
+The result of the diamond-square algorithm is a fractal with dimension D = 2 + H.
+
+A similar algorithm, midpoint displacement *TODO link to mpd in docs*, almost
+identically, except that in DiamondSquare, the ed
+"""
+struct DiamondSquare <: NeutralLandscapeMaker
+    H::Float64
+    function DiamondSquare(H::T) where {T <: Real}
+        @assert 0 < H && H < 1
+        new(H)
+    end
+    DiamondSquare() = new(0.5)
+
+end
+
+"""
+    MPD()
+
+    Creates a midpoint-displacement algorithm object `MPD`. The degree of spatial autocorrelation is controlled by a parameter `H`,
+    which varies from 0.0 (low autocorrelation) to 1.0 (high autocorrelation) --- note this is non-inclusive and H = 0 and H = 1 will not behavive as expected.
+"""
+struct MPD <: NeutralLandscapeMaker
```

Comment:
`MidPointDisplacement` would be better -- no other are abbreviated, and it's immediately obvious what it does